### ISSUE TITLE
[Bug]: App Store Connect threatens to block OneSignal apps

### ISSUE THREAD

**Issue by DaveBrask** [OneSignal/OneSignal-DotNet-SDK#127]
### What happened?

App Store Connect sent an email threatening to block OneSignal apps:

ITMS-91061: Missing privacy manifest - Your app includes “Frameworks/OneSignalCore.framework/OneSignalCore”, which includes OneSignalCore, an SDK that was identified in the documentation as a privacy-impacting third-party SDK. Starting November 12, 2024, if a new app includes a privacy-impacting SDK, or an app update adds a new privacy-impacting SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest. For more details about this policy, including a list of SDKs that are required to include signatures and manifests, visit: https://developer.apple.com/support/third-party-SDK-requirements.

ITMS-91061: Missing privacy manifest - Your app includes “Frameworks/OneSignalOutcomes.framework/OneSignalOutcomes”, which includes RxSwift, an SDK that was identified in the documentation as a privacy-impacting third-party SDK. Starting November 12, 2024, if a new app includes a privacy-impacting SDK, or an app update adds a new privacy-impacting SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest. For more details about this policy, including a list of SDKs that are required to include signatures and manifests, visit: https://developer.apple.com/support/third-party-SDK-requirements.

### Steps to reproduce?

```Markdown
Submit to App Store Connect for review
```

### What did you expect to happen?

I didn't expect the app store to object.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bnoffer** [comment]
I presume this is due to the DotNet SDK not yet being updated with the latest iOS Frameworks. Due to a incompatibility with XCode 16 in combination with .NET 8.0.403 I manually built the DotNet SDK against the OneSignal.XCFramework v. 5.2.5 which resolved the compilation issues I had and it passed the App Store without any issues. Therefore I would say this post is related to my post #125 .

**Comment by Marioo1357** [comment]
@bnoffer did you just make a bindings for both platforms?

**Comment by dombartenope** [comment]
The .NET SDK is a wrapper on top of the [iOS Native SDK](https://github.com/OneSignal/OneSignal-iOS-SDK/releases?page=2) where these privacy manifests were added in native release v5.2.0 

This should be resolved by updating the .NET SDK to the same version that supports the iOS release that contains these privacy manifests [v5.2.0](https://github.com/OneSignal/OneSignal-DotNet-SDK/releases/tag/5.2.0)

**Comment by bnoffer** [comment]
I used the .NET SDK v. 5.2.1 before running into the issues with iOS 18 and yes it worked fine with iOS 17. Therefore it was not an issue with the privacy manifests which were introduced with iOS 17 by the way. As mentioned above manually updating the iOS SDK to v. 5.2.5 resolved the issue. I have seen that you started work on the .NET SDK v. 5.2.2 which pushes the iOS SDK to v. 5.2.9, that should fix the issue for everyone. Any estimate when this will release? Because I would like not having to use my custom build in production any longer then I have to.

**Comment by IainS1986** [comment]
Is there any news on this? Specifically when the 5.2.2 release is due?

Is the only 'easy' way to work around this dropping back to XCode 15.4 and hoping this gets resolved before the April requirement to target iOS 18 :/ ?

**Comment by daltzctr** [comment]
This has unfortunately impacted my application with Apple **rejecting** the binary due to the aforementioned issues. Previously it was just a warning.

**Comment by IainS1986** [comment]
We are building with,

    <PackageReference Include="OneSignalSDK.DotNet" Version="5.2.1" />

With DevOps using

XCode 16.1
.Net 8 with workloads from 8.0.61 ('install ios --from-rollback-file https://github.com/dotnet/maui/releases/download/8.0.61/8.0.61.json --source https://api.nuget.org/v3/index.json')

And we aren't getting any warnings or issues from AppStore (we had a build just get approved a few days ago)

**Comment by daltzctr** [comment]
@IainS1986 We are targeting iOS 18 and XCode 16.3 with .NET 9.

**Comment by IainS1986** [comment]
@daltzctr Fair enough, we've kicked the .Net9 can down the road for a while (to the point we might just wait for .Net10)

But - we are building with XCode16.1 against iOS 18.1 - so not sure if that combo with .Net9 would be fine? 🤷

**Comment by Marioo1357** [comment]
@IainS1986 tried on 16.1 - same issue.
As I remember problem started after .Net update, not xcode 

The thing is that maui for .Net 8 is out of support. In Maui there is no lts 

I made my own workaround by binding your ios lib by myself. The only one problem is that in my workaround the trimming made app crash at start

**Comment by IainS1986** [comment]
Interesting, annoyingly getting OneSignal to ever respond or comment on issues on GitHub is like pulling teeth. 

I've had more success in the past via their support email. Might want to mail them (and then if you do, maybe update the resolution here because they also aren't great at that)

**Comment by Marioo1357** [comment]
I've tried on discord. The bad thing is that they removed Channel for Maui. The scond bad thing is that its hard to find info is .Net Maui abonded by them.
But Yes, I will try rękach them via email
